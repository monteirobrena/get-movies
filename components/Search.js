import React, { Component } from 'react';
import CssColors from './CssColors';
import { Button } from 'react-native-elements';
import { Actions } from "react-native-router-flux";
import { Text, TextInput, ListView, View, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  input: {
    height: 50,
    width: '75%',
    marginTop: 10,
    marginBottom: 20,
    borderBottomWidth: 2,
    borderBottomColor: CssColors.darkColor
  },
  text: {
    fontSize: 20,
    marginTop: 10,
    color: CssColors.darkColor
  },
  textError: {
    color: CssColors.errorColor
  },
  view: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: CssColors.lightColor
  }
});

export default class Search extends Component {
  constructor(props) {
    super(props);
    this.state = { showNotFound: false };
  }

  handlePress = () => {
    let textSearch = this.state.textSearch;
    let urlApiMovies = "https://jsonmock.hackerrank.com/api/movies/search/?Title=" + textSearch;

    return fetch(urlApiMovies)
      .then((response) => response.json())
      .then((responseJson) => {
        let data = responseJson.data;

        if (data.length > 0) { 
          this.setState({showNotFound: false});

          let ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});

          Actions.searchResults({ data: ds.cloneWithRows(data) });
        } else {
          this.setState({showNotFound: true});
        }
      })
		.catch((error) => {
			console.error(error);
		});
  }

  render() {
    return (
      <View style={styles.view}>
        { this.state.showNotFound && 
          <Text style={styles.textError}>
            Nenhum filme encontrado :(
          </Text>
        }
        <Text style={styles.text}>
          Sobre qual filme vocÃª quer saber mais?
        </Text>
        <TextInput
          style={styles.input}
          ref={(el) => { this.textSearch = el; }}
          onChangeText={(textSearch) => this.setState({textSearch})}
        />
        <Button
          large
          raised
          title="BUSCAR"
          icon={{name: 'search'}}
          backgroundColor={CssColors.primaryColor} 
          onPress={() => this.handlePress()}
        />
      </View>
    );
  }
}

