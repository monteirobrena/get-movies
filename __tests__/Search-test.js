import 'react-native';
import React, { Component } from 'react';
import { Button } from 'react-native-elements';
import { Text, TextInput, ListView, View, StyleSheet } from 'react-native';
import Search from '../components/Search';
import renderer from 'react-test-renderer';

const tree = renderer.create(
  <Search />
).toJSON();
 
it('renders correctly', () => {
  expect(tree).toMatchSnapshot();
});
