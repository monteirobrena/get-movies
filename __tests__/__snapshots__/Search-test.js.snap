// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f1f1f1",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#141414",
        "fontSize": 20,
        "marginTop": 10,
      }
    }
  >
    Sobre qual filme vocÃª quer saber mais?
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    style={
      Object {
        "borderBottomColor": "#141414",
        "borderBottomWidth": 2,
        "height": 50,
        "marginBottom": 20,
        "marginTop": 10,
        "width": "75%",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        Object {
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "backgroundColor": "#ff5722",
            },
            undefined,
            false,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                false,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          BUSCAR
        </Text>
      </View>
    </View>
  </View>
</View>
`;
